src/java.desktop/macosx/native/libawt_lwawt/awt/CPrinterJob.m

```diff
@@ -538,6 +539,30 @@ static void javaPrinterJobToNSPrintInfo(JNIEnv* env, jobject srcPrinterJob, jobj
             [dst updateFromPMPrintSettings];
         }
     }
+
+    jint chromaticity = (*env)->CallIntMethod(env, srcPrinterJob, jm_getChromaticity);
+    CHECK_EXCEPTION();
+
+    if (chromaticity >= 0) {
+
+        jobject chromaticityKeyValues = (*env)->CallObjectMethod(env, srcPrinterJob, jm_getChromaticityKeyValues, chromaticity);
+
+        if (chromaticityKeyValues == nil) {
+
```

Comment:
I'm not trying to revive this PR, but I was looking at it and I got curious about a couple of things.
What is going on here ?
First the Java code will never (I think) return NULL, so the "if" will never be used.
Second, why in the case that it does, did you decide to set ColorModel=Gray or ColorModel=Color ?
Are these a best guess default ? In which case why don't you always specify them anyway ?
You are already throwing the entire set of ones you read from the conf file into the settings.

Also there's no way I can see for you to KNOW that the settings will work. 
This could be confusing for apps wanting monochrome/gray 
Since you don't report back the supported status then they can't tell that when monochrome would work.

and (btw) you don't even filter this by whether the printer supports color when setting Color.

And on a broader note, the entire PPD API and PPD printer drivers are deprecated by CUPS.
So this seems like an approach on which the clock is ticking before it is even deployed.